include/complex_quda.h

```diff
@@ -511,6 +519,14 @@ struct complex <float> : public cuFloatComplex
       return *this;
     }
 
+  __host__ __device__
```

Comment:
Got it.

On 26.04.2016, at 17:54, maddyscientist <notifications@github.com<mailto:notifications@github.com>> wrote:


In include/complex_quda.h<https://github.com/lattice/quda/pull/458#discussion_r61110595>:

> @@ -511,6 +519,14 @@ struct complex <float> : public cuFloatComplex
>        return *this;
>      }
>
> +  __host__ __device__


I don't think these can be further templated. Note that there is a generic complex<ValueType> base class that implements operations like -= and *= etc., but there are also the specializations complex<float> and complex<double> that build upon cuComplex.h. These latter specializations need to be fully reimplemented, hence the functions you refer to are written using explicit float and double types. Agree?

-
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub<https://github.com/lattice/quda/pull/458/files/9b831966c56553288fbe635a0924020203b7f445#r61110595>

NVIDIA GmbH, Wuerselen, Germany, Amtsgericht Aachen, HRB 8361
Managing Director: Karen Theresa Burns

-----------------------------------------------------------------------------------
This email message is for the sole use of the intended recipient(s) and may contain
confidential information.  Any unauthorized review, use, disclosure or distribution
is prohibited.  If you are not the intended recipient, please contact the sender by
reply email and destroy all copies of the original message.
-----------------------------------------------------------------------------------